### DISCUSSION TITLE
Drag object

### DISCUSSION THREAD

**Discussion by ricardoekm** [visgl/deck.gl#5506] [Q&A]
Hi,
We'd like to allow the user to drag objects in a scatterplot plot. Is there anything in the [event manager](https://github.com/visgl/deck.gl/blob/master/dev-docs/RFCs/v4.1/event-handling-rfc.md) that could help us achieve this (e.g. drag object started)? What would be the best way of handling this in terms of event handling?

In terms of the layer, I guess there's no alternative other than updating the data as the user drags and re-render the layer, am I correct?

Any guidance would be appreciated. Thanks!

**Comment by Pessimistress** [comment]
There are `onDragStart` `onDrag` and `onDragEnd` callbacks in [layer props](https://deck.gl/docs/api-reference/core/layer#interaction-properties).

Here is a simple example using `updateTriggers.getPosition` to update the rendering during dragging: https://codepen.io/Pessimistress/pen/eYdmPjJ?editors=0010

If you have a lot of objects in one layer, you will need to look into [partial data update](https://deck.gl/docs/api-reference/core/layer#_datadiff).

**Comment by ibgreen** [comment]
For serious editing, you may also want to consider using the editable deck.gl layers in https://nebula.gl/.

**Comment by ricardoekm** [comment]
Awesome, thanks for the tips guys.
